### PR TITLE
Improve ci config

### PR DESCRIPTION
**Pull request by cprussin**
A few related things:

- Add nix shell configuration
- Improve CI configuration
- Fix some type errors
- Fix prettier format issues

Usually I'd separate these to separate commits but all but the last were pretty minimal.  I split the prettier format fixes to a separate commit in case you want to review just the first 3 separately.

### CODE REVIEW

**Review by cctdaniel — APPROVED** [review]
looks good but can you fix the failing test before merging?

**Comment by cprussin** [comment]
> looks good but can you fix the failing test before merging?

of course, working on it and will merge once fixed

**Pull request merged**
